test/unit/media.js

```diff
@@ -250,4 +254,54 @@ test('should handle unsupported sources with the source hanlder API', function()
 
   tech.setSource('');
   ok(usedNative, 'native source handler was used when an unsupported source was set');
-});
\ No newline at end of file
+});
+
+test('should emulate the video element\'s behavior for currentSrc when src is set', function(){
+  var mockPlayer = {
+    off: this.noop,
+    trigger: this.noop
+  };
+  var sourceA = { src: 'foo.mp4', type: 'video/mp4' };
+  var sourceB = { src: '', type: 'video/mp4' };
+
+  // Define a new tech class
+  var Tech = videojs.MediaTechController.extend();
+
+  // Extend Tech with source handlers
+  vjs.MediaTechController.withSourceHandlers(Tech);
+
+  // Create an instance of Tech
+  var tech = new Tech(mockPlayer);
+
+  // Create source handlers
+  var handler = {
+    canHandleSource: function(source){
+      return 'probably';
+    },
+    handleSource: function(s, t){return {};}
+  };
+
+  Tech.registerSourceHandler(handler);
+
+  // Pass a source through the source handler process of a tech instance
+  tech.setSource(sourceA);
+
+  // Test that currentSource_ is not immediately specified
+  strictEqual(tech.currentSource_, undefined, 'sourceA was not stored immediately');
```

Comment:
The current implementation is backwards compatible with existing techs. Moving anything to MediaController either doesn't buy us code savings or requires potentially breaking changes which we shouldn't be making in VJS4.

I agree that for VJS5 we should strongly consider moving much the `currentSrc` function to `Tech` and have each child-tech implement something like a `currentSrc_` function that `Tech#currentSrc` will look for and call if `currentSource_` is not defined.